### ISSUE TITLE
Setting icon title with printf and autocommand flashes screen

### ISSUE THREAD

**Issue by DieSpinne** [vifm/vifm#587]
I have set `autocmd DirEnter '*' :!printf '\033]1;'"$(dirs)"'\007'` in my configuration file, so that the icon title (which is the title that shows in my taskbar) of the window is always updated with the active panel's directory name. It works. However, XTerm's screen flashes every time that command is triggered. Ignoring the command output with `%i` makes the title change stop working, as well as backgrounding it. Unsetting `title` option makes no difference. I tested the same in Terminator, and although the problem does not happen there, the titles also do not change, maybe it does not support it.

Any ideas to workaround this (if it is possible at all)? Thank you!

PS: Notice the window title is correct when `title` is set, the misbehavior I'm asking about is with the icon title that shows up in the taskbar.

**Comment by xaizek** [comment]
Try:
```vim
autocmd DirEnter '*' :!printf '\033]1;'"$(dirs)"'\007' > /dev/tty %i
```

**Comment by DieSpinne** [comment]
Many thanks, that works exactly as I expected. Should you want, this issue can be closed.

**Comment by xaizek** [comment]
I thought that setting "Icon Name" in the [documentation](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h3-Operating-System-Commands) is a way to pick an icon... That's why Vifm is using `\e]2;` sequence. Looks like it can be changed to `\e]0;` then, thanks. Will do this after 0.11 release to give time to test it with various terminals.

**Comment by DieSpinne** [comment]
May I suggest that `\e]2;` is left as is for the window title and `\e]1;` for the icon title is configured by another setting, such that they can be kept different? This would be the analogous of Vim's `titlestring` and `iconstring`. The reason is that in the taskbar the string needs to be kept short, while the window title can be large without truncation (in a maximized window).

**Comment by xaizek** [comment]
Then I guess it's pointless to change anything without adding `'titlestring'` and `'iconstring'`.

**Comment by DieSpinne** [comment]
Indeed. Personally, I don't mind having that `printf` trick, but we have to admit it is not quite user friendly :)

**Comment by DieSpinne** [comment]
Looking at it again, I reckon that as long as one adds this to `vifmrc`
```
autocmd DirEnter '*' :!printf '\033]2;'"Title:$(dirs)"'\007' > /dev/tty %i
autocmd DirEnter '*' :!printf '\033]1;'"Icon:$(dirs)"'\007' > /dev/tty %i
```
there is very small advantage (besides user-friendliness) in adding `titlestring` and `iconstring`.

**Comment by xaizek** [comment]
One advantage is updating titles when you switch between panes or tabs. `DirEnter` isn't run for those cases.
